packages/api/src/add-build.js

```diff
@@ -60,11 +60,14 @@ function buildPassingPercent (successes, failures) {
 async function addBuild (testCases, buildInfo, client, collectionName = 'repositories-fake') {
   var dbRepo = client.collection(collectionName).doc(buildInfo.repoId);
 
-  // if repo does not exist, add information
-  await dbRepo.update({
-    organization: buildInfo.organization,
-    url: buildInfo.url
-  }, { merge: true });
+  // if repo does not exist, add information including environment variables and possibilities
+  const repoUpdate = { };
+  for (const prop in buildInfo.environment) {
+    repoUpdate['environments.' + prop] = Firestore.FieldValue.arrayUnion(buildInfo.environment[prop]);
```

Comment:
are there any illegal characters we should be concerned about, what happens for instance if the property is `foo.bar`, will we accidentally create an extra level of nesting in FireStore?

I wonder if `Firestore` provides a method for escaping strings, or has documentation on doing so.

Reply:
Fixed with firebaseEncode function in lib/firebase-encode.js and test/lib-test.js